src/commands/client/login.js

```diff
@@ -17,6 +19,17 @@ module.exports = async args => {
     return error(`Failed to extract login creds from: ${token}`);
   }
 
+  const res = await invokeCommand('cli_login ' + token, token);
+  if (!res.data || !res.data.text) {
+    return error(`Failed to login using creds: ${token}`);
+  }
+  const cmd = shell.exec('nim auth login --auth ' + res.data.text);
```

Comment:
why is a shell command needed in addition to the call to setCredentials (internal nim api) on line 21/34?

Reply:
Because it sets the `commander` specific creds under credentials.json and doesn't add the `namespace` creds.

Reply:
There is a function in the credential store to do that:

https://github.com/nimbella/nimbella-cli/blob/b7979517229f514be664117ec287988a74ed3bf9/deployer/src/credentials.ts#L77

Reply:
But i suppose you don't know the namespace thought at this point (although it can be fetched).

Reply:
I believe I understand how we may have gotten to this point.

It is not possible for commander-cli to use the command invocation API so, _if_ something needs to be done at that level you do have to spawn a process.   And, at one point, if you wanted to do a full login, you could not do that with the deployer API alone.

However, that has changed, so I think _now_ it should not be necessary to spawn.  You can use 

```
doLogin(token: string, persister: Persister, host: string = DEFAULT_API_HOST): Promise<Credentials>
```

The above function is exported by the deployer library.   It returns Credentials for convenience but it also stores them (it does the login).

Note that that _is_ a full login, using a token, but that's actually what you want because if you do a login using only `--auth` then you will not get full credentials.   So, the recommended flow (if all you have is an OW auth key) is (1) use the tokenizer API to convert that to a token, (2) use `doLogin` to fully login with that token (3) optionally add commander credentials with `addCommanderData` (you now know the namespace because it is in the returned Credentials).

Reply:
Thanks for the clarification and explanation. We should adopt this approach. I am neutral on merging this pr and following it up with another. Will wait got Nikhil’s input. This patch is reviewed/approved.

Reply:
I will work on this tomorrow as a separate PR. Merging this now